### PR TITLE
feat: Add checklist icons

### PR DESCRIPTION
**Pull request by ipiedrahb**
Adds checklist icons per this [story](https://app.shortcut.com/homebound-team/story/31258/add-checklist-column-to-schedule)

[Figma](https://www.figma.com/file/ePfj5WZKtGfOgABpZcUtRE/%E2%9C%A8Beam-Icons?node-id=1481-55&t=01pvRip2psC356xR-0)

**Figma Screenshots**
Screen Shot 2023-04-17 at 10 36 08
Screen Shot 2023-04-17 at 10 36 15

**Storybook**
Screen Shot 2023-04-17 at 10 58 32

### CODE REVIEW

File: src/components/Icon.stories.tsx [hunk 1]
```diff
@@ -181,7 +183,7 @@ export const Icon = (props: IconProps) => {
       <ul css={{ gap: 24, listStyle: "none", gridTemplateColumns: "repeat(4, 1fr)", ...Css.dg.p0.$ }}>
         {miscIcons.map((icon, i) => (
           <li css={{ gap: 8, ...Css.xsMd.df.aic.fdc.$ }} key={icon}>
-            <IconComponent icon={icon} data-testid={icon} id={icon} color={props.color} />
+            <IconComponent icon={icon} data-testid={icon} id={icon} color={props.color}/>
```
Comment (bdow) [inline src/components/Icon.stories.tsx:186]: **request:** Can you revert this change back to including that space? 
```suggestion
            <IconComponent icon={icon} data-testid={icon} id={icon} color={props.color} />
```

**Review by bdow — APPROVED** [review]

File: src/components/Icon.stories.tsx (hunk 1 above)
Comment (ipiedrahb) [reply src/components/Icon.stories.tsx]: @bdow 

Sure, thank you!
